ext/enchant/enchant.c

```diff
@@ -138,65 +162,71 @@ static void php_enchant_list_dicts_fn( const char * const lang_tag,
 }
 /* }}} */
 
-static void php_enchant_broker_free(zend_resource *rsrc) /* {{{ */
-{
-	if (rsrc->ptr) {
-		enchant_broker *broker = (enchant_broker *)rsrc->ptr;
-		if (broker) {
-			if (broker->pbroker) {
-				if (broker->dictcnt && broker->dict) {
-					if (broker->dict) {
-						int total;
-						total = broker->dictcnt-1;
-						do {
-							if (broker->dict[total]) {
-								enchant_dict *pdict = broker->dict[total];
-								broker->dict[total] = NULL;
-								zend_list_free(pdict->rsrc);
-								efree(pdict);
-							}
-							total--;
-						} while (total>=0);
-					}
-					efree(broker->dict);
-					broker->dict = NULL;
-				}
-				enchant_broker_free(broker->pbroker);
-			}
-			efree(broker);
+static void php_enchant_broker_free(zend_object *object) /* {{{ */
+{
+	enchant_broker *broker = enchant_broker_from_obj(object);
+
+	if (broker) {
```

Comment:
This check is unnecessary, from_obj results can never be NULL.

Reply:
fixed